addons/account_payment/static/src/js/payment_form.js

```diff
@@ -0,0 +1,56 @@
+/** @odoo-module **/
+
+import PaymentForm from '@payment/js/payment_form';
+
+PaymentForm.include({
+
+    /**
+     * Set whether we are paying an installment before submitting.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {Event} ev
+     * @return {void}
+     */
+    async _submitForm(ev) {
+        ev.stopPropagation();
+        ev.preventDefault();
+
+        const paymentDialog = this.el.closest('#pay_with');
+        const chosenPaymentDetails = paymentDialog ? paymentDialog.querySelector('.o_btn_payment_tab.active'): null;
+        if (chosenPaymentDetails && chosenPaymentDetails.id === 'o_payment_installments_tab') {
+            this.paymentContext.payNextInstallment = true;
+        }
+        await this._super(...arguments);
+    },
```

Comment:
I think this should rather be done in the `start`. Overrides of `_submitForm` are generally done to ensure all required inputs have been filled in.

Reply:
The customer can click on the "Installment" or "Full Amount" button when the dialog is launched, and on submitting it should get the currently selected button.